address.go

```diff
@@ -375,3 +483,82 @@ func (a *AddressPubKey) AddressPubKeyHash() *AddressPubKeyHash {
 func (a *AddressPubKey) PubKey() *btcec.PublicKey {
 	return a.pubKey
 }
+
+// AddressSegwit is an Address for a pay-to-witness-pubkey-hash (P2WPKH) or
```

Comment:
`p2wkh` and `p2wsh` are _only_ defined for version zero witness programs. I don't think we need to worry about the case of non-zero for now, as in that event a new BIP would need to be created in order to reach wide adoption.